File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 1
        n=len(prices)
        max_profit = 0

        while(r<n):
            profit = prices[r] - prices[l]
            max_profit = max(profit,max_profit)
            if(prices[l]>prices[r]):
                l=r
                r=l+1
                continue
            r+=1
        return max_profit
            
                    
        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.cpp
class Solution {
public:

    int characterReplacement(string s, int k) {
        //check when string is empty

        // start with maximum window size
        // check max frequency element
        // if total - max frequency <=k then all good else
        // while that condition is not true l++
        int n = s.size();
        map<char,int> mp;
        for(auto x : s){
            mp[x]++;
        }

        int result = 0;

        for(auto x:mp){
            int l = 0;
            int r = 0;
            int count = 0;
            for(r = 0; r<n;r++){
                if(s[r]==x.first){
                    count+=1;
                }
                while ( (r-l+1) - count > k){
                    if(s[l]==x.first){
                        count--;
                    }
                    l++;
                }
                result = max(result,r-l+1);
            }
        }
                    return result;

        
    }
};
